tests/TestCase/Database/ConnectionTest.php

```diff
@@ -355,10 +355,10 @@ public function testInsertWithPositionalTypes(): void
         $result->closeCursor();
         $this->assertInstanceOf('Cake\Database\StatementInterface', $result);
         $result = $this->connection->execute('SELECT * from things where id  = 3');
-        $this->assertCount(1, $result);
```

Comment:
I wonder how much userland code is dependent on statements being Countable? We might be able to add a deprecation in 4.x if we update the internals of the ORM to avoid using `count()`.

Reply:
Adding deprecation in 4.x might be possible, still would be a big chore to update all the tests.